llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.h

```diff
@@ -187,9 +187,12 @@ class GCNTTIImpl final : public BasicTTIImplBase<GCNTTIImpl> {
       }
       return false;
     }
-    if ((FromAS == AMDGPUAS::CONSTANT_ADDRESS_32BIT &&
-         ToAS == AMDGPUAS::CONSTANT_ADDRESS) ||
-        (FromAS == AMDGPUAS::CONSTANT_ADDRESS &&
+    if (FromAS != ToAS &&
+        (FromAS == AMDGPUAS::GLOBAL_ADDRESS ||
+         FromAS == AMDGPUAS::CONSTANT_ADDRESS ||
+         FromAS == AMDGPUAS::CONSTANT_ADDRESS_32BIT) &&
+        (ToAS == AMDGPUAS::GLOBAL_ADDRESS ||
+         ToAS == AMDGPUAS::CONSTANT_ADDRESS ||
          ToAS == AMDGPUAS::CONSTANT_ADDRESS_32BIT))
```

Comment:
The whole function should match isExtendedGlobalAddrSpace. We should accept arbitrary 64-bit address spaces as an alias for global. The part above under ToAS == AMDGPUAS::FLAT_ADDRESS is also too strict.

The only illegal cases involve region and anything, fat pointers and anything, or any of the 32-bit address spaces between each other.

Reply:
Would it make sense to have a table like for [the aliasing rules](https://github.com/llvm/llvm-project/blob/a3b0c31ebc2f6fe672f08f6b7d15f25a2b26edda/llvm/lib/Target/AMDGPU/AMDGPU.h#L473) for this, then?

Reply:
It's a stylistic choice, whatever comes out simpler

Reply:
Adjusted in the most recent commit.